jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/LexBreathFirstSearch.java

```diff
@@ -0,0 +1,217 @@
+        initialVertexList.remove(startingVertex);
+        
+        labelToVertexMap.put("", initialVertexList);
+        
+        for(V vertex : initialVertexList) {
+            vertexToLabelMap.put(vertex, "");
+        }
+        
+        // ... except for the starting vertex
+        ArrayList<V> startingVertexList = new ArrayList<V>();
+        startingVertexList.add(startingVertex);
+        
+        labelToVertexMap.put(maxLabel, startingVertexList);
+        vertexToLabelMap.put(startingVertex, maxLabel);
+        
+        for(int i = n; i >= 1; i--) {
+            // pick unvisited vertex v with largest label
+            ArrayList<V> candidates = labelToVertexMap.get(maxLabel);
+            V currentVertex = candidates.get(0);
+            
+            // set index of v to be n + 1 - i
+            ordering.add(currentVertex);
+            
+            // for every unvisited neighbor w of v, append i to label of w
+            for(V neighbor : Graphs.neighborListOf(graph, currentVertex)) {
+                String oldLabel = vertexToLabelMap.get(neighbor);
+                String newLabel = oldLabel + Integer.toString(i);
+                
+                // remove neighbor from previous list
+                ArrayList<V> oldList = labelToVertexMap.get(oldLabel);
+                oldList.remove(neighbor);
+                
+                // if there are no vertices left in this list, delete the entry
+                if(oldList.isEmpty()) {
+                    labelToVertexMap.remove(oldLabel);
+                } else {
+                    // else we replace the list with its updated version 
+                    labelToVertexMap.replace(oldLabel, oldList);
+                }
+                
+                // now add neighbor to the list of the new label
+                if(labelToVertexMap.containsKey(newLabel)) {
+                    ArrayList<V> newList = labelToVertexMap.get(newLabel);
+                    newList.add(neighbor);
+                    labelToVertexMap.replace(newLabel, newList);
+                } else {
+                    ArrayList<V> newList = new ArrayList<V>();
+                    newList.add(neighbor);
+                    labelToVertexMap.put(newLabel, newList);
+                }
+                
+                // associate neighbor with its new label
+                vertexToLabelMap.replace(neighbor, newLabel);
+            }
+            
+            // mark current vertex as visited, i.e. remove its entries from the maps
+            vertexToLabelMap.remove(currentVertex);
+            candidates.remove(currentVertex);
+            labelToVertexMap.replace(maxLabel, candidates);
+        }
+    
+        // return ordering -- thanks Java for not being able to create an array from a list with generic types for some reason
+        V[] orderingArray = null;
+        orderingArray = ordering.toArray(orderingArray);
```

Comment:
What this does is copying the entire internal array again. Maybe it would be best to not use ArrayList at all and work on a basic array directly, especially because you dont need any feature of the list itself and you know the capacity at the start.